### ISSUE TITLE
`pkg install` fails with "WASI Exit error: 1"

### ISSUE THREAD

**Issue by kkebo** [holzschu/a-shell#799]
clang doesn't work, and also I can't install it with `pkg`.

```console
$ which clang
/private/var/containers/Bundle/Application/27AAD37A-A2B0-4771-A5F8-C5179A5FF8A3/a-Shell.app/bin/clang
$ clang
In order to use clang, you need to install or update the C SDK with 'pkg install llvm-18'.
$ lld
In order to use lld, you need to install or update the C SDK with 'pkg install llvm-18'.
$ pkg install llvm-18
WASI Exit error: 1
Package llvm-18 not found
```

- a-Shell 1.15.1 (416) (installed from TestFlight)
- iPadOS 18 Developer Beta 3 (22A5307i)

**Comment by holzschu** [comment]
There's something weird with `pkg install llvm-18` returning `WASI Exit error: 1`. `pkg`  is a shell command, not a WebAssembly command. Unless you have another `pkg` command in your path?

**Comment by kkebo** [comment]
No, I don't have any other `pkg` commands.

```console
$ which -a pkg
/private/var/containers/Bundle/Application/27AAD37A-A2B0-4771-A5F8-C5179A5FF8A3/a-Shell.app/bin/pkg
$ find $HOME/Documents/bin -name "pkg*"
$ file "/private/var/containers/Bundle/Application/27AAD37A-A2B0-4771-A5F8-C5179A5FF8A3/a-Shell.app/bin/pkg"
/private/var/containers/Bundle/Application/27AAD37A-A2B0-4771-A5F8-C5179A5FF8A3/a-Shell.app/bin/pkg: POSIX shell script, ASCII text executable
```

**Comment by holzschu** [comment]
Thanks for the quick reply. Still, a-Shell should not write "WASI exit error" unless it is trying to execute a wasm command.
What's more puzzling is that it breaks down before the "echo Downloading $packagename" on line 66 of pkg, which is very early in the process. 

For now, I have no idea what is happening. I assume it breaks the same way for other packages?

**Comment by kkebo** [comment]
So far this has not happened except in `pkg`. I'll continue to investigate it, and report here if I find anything, Thank you.

**Comment by kkebo** [comment]
Here is my `bin` dir in case you are wondering.

```console
$ ls bin
base64                  getopt                  mktemp                  seq                     swift-format.wasm       which                   zipnote.wasm
funzip.wasm             hd                      od                      sqlite3.wasm            unzip.wasm              zip.wasm                zipsplit.wasm
base64                  getopt                  mktemp                  seq                     swift-format.wasm       which                   zipnote.wasm
funzip.wasm             hd                      od                      sqlite3.wasm            unzip.wasm              zip.wasm                zipsplit.wasm
gawk                    hexdump                 rg                      swift-format-main.wasm  unzipsfx.wasm           zipcloak.wasm
```

**Comment by kkebo** [comment]
```console
$ cat .profile
export LANG=en_US.UTF-8
export EDITOR=vim
export XDG_CONFIG_HOME=$HOME/Documents/.config

# For dash
export PS1='$(echo "\\033[48;2;76;86;106m") $PWD $(echo "\\033[m\\033[38;2;76;86;106m\\033[m ")'

alias view="vim -RM"
alias vi="vim"

alias ll="ls -lh"
alias la="ls -lah"

if test -e rg; then
    alias grep="rg"
fi

if test -e config; then
    config -s 12 -n 'SFMono Nerd Font'
fi
$ cat .bashrc 
# For a-Shell
export PS1='\e[48;2;76;86;106m \w \e[m\e[38;2;76;86;106m\e[m '
```

**Comment by kkebo** [comment]
It turned out that `alias grep="rg"` was the cause.

**Comment by holzschu** [comment]
Of course! (although that raises the other question why `rg` fails inside a Shell script). I should probably make `pkg` more robust by ensuring it calls the `$APPDIR/bin` version of the commands instead of something else.

**Comment by holzschu** [comment]
By the way, I have uploaded a new version of the llvm-18 package, with a few precompiled libraries included (libz, libpng, libexpat, libfreetype). And the next version of a-Shell fixes an issue with setjmp/longjmp in clang (uploaded tonight, active tomorrow).

**Comment by RadioNoiseE** [comment]
> By the way, I have uploaded a new version of the llvm-18 package, with a few precompiled libraries included (libz, libpng, libexpat, libfreetype). And the next version of a-Shell fixes an issue with setjmp/longjmp in clang (uploaded tonight, active tomorrow).

Does it mean we can then use setjmp.h?

**Comment by holzschu** [comment]
Yes! 
You need to add: -mllvm -wasm-enable-sjlj  to the C flags, and  -lsetjmp to the ld flags.
By the way, since configure does not work well on the iPad, you can also download the compiler and SDK for OSX here: https://github.com/holzschu/wasi-sdk/releases/tag/wasi-sdk-aShell-19

**Comment by kkebo** [comment]
I have confirmed that this was fixed in 1.15.1 (417).
